convert2rhel/subscription.py

```diff
@@ -277,28 +287,131 @@ def args(self):
 
         return args
 
-    def __call__(self):
+    @property
+    def connection_opts(self):
         """
-        Run the subscription-manager command.
+        This property is a dbus.Dictionary that contains the connection options for RHSM
+        dbus calls.
 
-        Wrapper for running the subscription-manager command that keeps
-        secrets secure.
+        Set :attr:`server_url` to affect this value.
         """
-        if self.password:
-            loggerinst.debug(
-                "Calling command '%s %s'" % (self.cmd, " ".join(hide_secrets(self.args)))
-            )  # lgtm[py/clear-text-logging-sensitive-data]
-            output, ret_code = utils.run_cmd_in_pty(
-                [self.cmd] + self.args, expect_script=(("[Pp]assword: ", self.password + "\n"),), print_cmd=False
-            )
-        else:
-            # Warning: Currently activation_key can only be specified on the CLI. This is insecure
-            # but there's nothing we can do about it for now. Once subscription-manager issue:
-            # https://issues.redhat.com/browse/ENT-4724 is implemented, we can change both password
-            # and activation_key to use a file-based approach to passing the secrets.
-            output, ret_code = utils.run_subprocess([self.cmd] + self.args, print_cmd=False)
+        connection_opts = {}
+
+        url_parts = None
+        if self.server_url:
+            try:
+                url_parts = parse_url(self.server_url)
+            except ValueError:
+                # url_parts will remain None which will mean we return an empty dbus.DIctionary
+                pass
```

Comment:
Untested

Reply:
This branch is now tested.  But we're looking into some changes to this behaviour (error instead of pass).